widget/list.go

```diff
@@ -210,6 +223,43 @@ func (l *List) UnselectAll() {
 	}
 }
 
+func (l *List) visibleItemHeights(itemHeight float32, length int) (visible map[int]float32, offY float32, minRow, maxRow int) {
```

Comment:
This doesn't seems ideal in term of performance as it will force a reallocation, recalculate and freeing every time the list is updated. It would be better if the map was allocated as part of the widget structure and just updated if the position of the visible item change.

Reply:
I just copied the code that has been in table for the last major release.
Do we need to review that as well?

Reply:
There is a change, as far as I can see, compared to the previous code, it is now iterating over the table which it wasn't before. If I understand it correctly, it is starting from zero and might break early, so as your list of content get longer and you are displaying the end of the list, this start to be a more costly operation which will impact current users even if they do not use the custom height. Minimizing the impact of switching from a O(1) algorithm to an O(n) seems like a good idea here. My understanding is that this won't be called as often as MinSize, so I will be ok if we delay optimizing this part, but it would be good to take note that rendering the list will now be a function of the number of items which wasn't the case before.